miio/integrations/chunmi/cooker_multi/cooker_multi.py

```diff
@@ -0,0 +1,400 @@
+
+    @property
+    @sensor("Menu ID")
+    def menu(self) -> str:
+        """Selected menu id."""
+        try:
+            return COOKING_MENUS[self.data["menu"]]
+        except KeyError:
+            return "Unknown menu"
+
+    @property
+    @sensor("Cooking stage")
+    def stage(self) -> str:
+        """Current stage if cooking."""
+        try:
+            return COOKING_STAGES[self.data["phase"]]["name"]
+        except KeyError:
+            return "Unknown stage"
+
+    @property
+    @sensor("Current temperature", unit="C")
+    def temperature(self) -> Optional[int]:
+        """Current temperature, if idle.
+
+        Example values: 29
+        """
+        return self.data["temp"]
+
+    @property
+    @sensor("Cooking finish time")
+    def ready_at(self) -> datetime.datetime:
+        """Remaining minutes of the cooking process."""
+        return (
+            datetime.datetime.now()
+            + datetime.timedelta(
+                0, int(self.data["t_left"] + (self.data["t_pre"] * 60))
+            )
+        ).replace(second=0, microsecond=0)
+
+    @property
+    @sensor("Cooking start delay")
+    def cooking_delayed(self) -> Optional[int]:
+        """Wait n minutes before cooking / scheduled cooking."""
+        delay = int(self.data["t_pre"])
+
+        if delay >= 0:
+            return delay
+
+        return None
+
+    @property
+    @sensor("Cooking duration")
+    def duration(self) -> int:
+        """Duration of the cooking process."""
+        return int(self.data["t_cook"])
+
+    @property
+    @sensor("Keep warm after cooking enabled")
+    def keep_warm(self) -> bool:
+        """Keep warm after cooking?"""
+        return self.data["akw"] == 1
+
+    @property
+    @sensor("Taste ID")
+    def taste(self) -> None:
+        """Taste id."""
+        return self.data["taste"]
+
+    @property
+    @sensor("Rice ID")
+    def rice(self) -> None:
+        """Rice id."""
+        return self.data["rice"]
+
+    @property
+    @sensor("Selected favorite recipe")
+    def favorite(self) -> None:
+        """Favored recipe id."""
+        return self.data["favs"]
+
+
+class MultiCooker(Device):
+    """Main class representing the multi cooker."""
+
+    _supported_models = [*MODEL_MULTI]
+
+    @command()
+    def status(self) -> CookerStatus:
+        """Retrieve properties."""
+        properties = [
+            "status",
+            "phase",
+            "menu",
+            "t_cook",
+            "t_left",
+            "t_pre",
+            "t_kw",
+            "taste",
+            "temp",
+            "rice",
+            "favs",
+            "akw",
+            "t_start",
+            "t_finish",
+            "version",
+            "setting",
+            "code",
+            "en_warm",
+            "t_congee",
+            "t_love",
+            "boil",
+        ]
+
+        values = []
+        for prop in properties:
+            values.append(self.send("get_prop", [prop])[0])
```

Comment:
I suppose the device doesn't support querying multiple properties at once? Just wondering as this can be a rather slow operation depending on how responsive the device is..

Reply:
When I try to query multiple properties at once the response is only containing the value for the first property queried. I don't really know if this can be investigated further but I guess it's not possible.

Reply:
Okay, it happens on some devices so that's fine. If you run `pre-commit run -a` locally, it should fix the linting issues.